dubbo-config/dubbo-config-api/src/main/java/com/alibaba/dubbo/config/ReferenceConfig.java

```diff
@@ -423,7 +427,14 @@ private T createProxy(Map<String, String> map) {
         if (c && !invoker.isAvailable()) {
             // make it possible for consumer to retry later if provider is temporarily unavailable
             initialized = false;
-            throw new IllegalStateException("Failed to check the status of the service " + interfaceName + ". No provider available for the service " + (group == null ? "" : group + "/") + interfaceName + (version == null ? "" : ":" + version) + " from the url " + invoker.getUrl() + " to the consumer " + NetUtils.getLocalHost() + " use dubbo version " + Version.getVersion());
+            final String serviceKey = (group == null ? "" : group + "/") + interfaceName + (version == null ? "" :

```

Comment:
Here you can use `BaseServiceMetadata.buildServiceKey(path, group, version);`

Reply:
@BurningCN  Thanks. But you may see this is 2.6.x version